### DISCUSSION TITLE
Supporting non-explicit arithmetic operations

### DISCUSSION THREAD

**Discussion by jrozner** [Vector35/binaryninja-api#4286] [Q&A] [answered]
This issue https://github.com/Vector35/binaryninja-api/issues/2043 covers de-optimizing math to clarify optimizations in the decompliation. There are a number of embedded architectures that lack certain arithmetic operations (eg. msp430 lacks multiplication and division). Is this a good solution for solving that as well or is there a better alternative? Compiler specific workflows to rewrite the IL?

**Comment by plafosse** [comment, marked as answer]
Yes compiler/platform specific workflows would be the way we could handle these types of cases.
